frontend/src/components/searchBar/useMessageSearchBar.tsx

```diff
@@ -4,13 +4,22 @@ import { useSearchParams } from "react-router-dom";
 const useMessageSearchBar = () => {
     const [searchParams, setSearchParams] = useSearchParams();
 
+    const searchMode = searchParams.get("searchMode") || "message";
+
     const handleSetSearchParams = (event: ChangeEvent<HTMLInputElement>) => {
-        setSearchParams({
-            messagesFilter: event.target.value
-        });
+        searchParams.set("messagesFilter", event.target.value);
+        setSearchParams(searchParams);
+    };
+
+    const handleSetSearchMode = () => {
```

Comment:
Could be simplify, set value from arugent, then 'if' is unnecessary.

And for what u use Params.set and setParams, u can't use only one of this?

Reply:
```ts
const setMenuComponent = (params: { [key in MenuPosition]?: string }) => {
    const allParams: Record<string, string> = {};

    searchParams.forEach((value, key) => {
      allParams[key] = value;
    });

    setSearchParams({ ...allParams, ...params });
  };
```
@masterHAWK99 Look that's example above, use object inside setter, that's my thought, here's code.